### PR TITLE
[E2E] add udf manage test

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

#7312

Since I can't find a way to skip the prompt of downloading jar files from chromium. This PR does not include download testing.

### CODE REVIEW

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+    private static final String user = "admin";
+
+    private static final String password = "dolphinscheduler123";
+
+    private static final String email = "admin@gmail.com";
+
+    private static final String phone = "15800000000";
+
+    private static final String testDiretoryName = "test_directory";
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java]: typo `testDiretoryName` --> `testDirectoryName`, I remember there is another typo in the other test case (FileManage?), can you also fix that?

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    private static final String testRenameDirectoryName = "test_rename_directory";
+
+    private final String testUploadUdfFilePath = Objects.requireNonNull(this.getClass().getClassLoader().getResource("docker/udf-manage/hive-jdbc-3.1.2.jar")).getPath();
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java]: You add a binary file (`.jar`) in the codebase, when releasing source codes it's also included, which is not allowed in ASF policy. You have some choices:

1. Exclude the `.jar` file in the release script, but the tests can be executed successfully when people download the source tars from our releases.
2. Download the jar on the fly, just like what you did for mysql datasource.
3. Provide the source codes of the jar, and build/package to generate the `jar` file before running this test.

If the jar file is randomly picked from the internet, choice 2 is preferable

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java]: I'll fix it.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        URL url = new URL(downloadUrl);
+
+        URLConnection conn = url.openConnection();
+        InputStream inStream = conn.getInputStream();
+        FileOutputStream fs = new FileOutputStream(filePath);
+
+        byte[] buffer = new byte[1024];
+        while ((byteRead = inStream.read(buffer)) != -1) {
+            byteSum += byteRead;
+            System.out.println(byteSum);
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java]: Remove this or replace with a logger

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            .goToTab(UserPage.class)
+            .update(user, user, password, email, phone)
+            .goToNav(ResourcePage.class)
+            .goToTab(UdfManagePage.class);
+    }
+
+    @AfterAll
+    @SneakyThrows
+    public static void cleanup() {
+        Files.walk(Constants.HOST_CHROME_DOWNLOAD_PATH)
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java:99]: Can you also clean up `testUploadUdfFilePath`?

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+    void downloadFile(String downloadUrl, String filePath) throws Exception {
+        int byteSum = 0;
+        int byteRead;
+
+        URL url = new URL(downloadUrl);
+
+        URLConnection conn = url.openConnection();
+        InputStream inStream = conn.getInputStream();
+        FileOutputStream fs = new FileOutputStream(filePath);
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java:182]: The 2 resources should be closed

**Review by kezhenxu94 — COMMENTED** [review]
Generally LGTM, only some nits

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java (hunk 5 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/UdfManageE2ETest.java:182]: I was too careless. I'll fix it.

**Review by kezhenxu94 — DISMISSED** [review]

**Comment by kezhenxu94** [comment]
@SbloodyS can you check the failed case?

**Comment by SbloodyS** [comment]
> @SbloodyS can you check the failed case?

I'm checking.

**Comment by SbloodyS** [comment]
@kezhenxu94 Add a page refresh in ```testCreateDirectory```. Please review again. Thanks.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**